homer0/projext-plugin-webpack-angularjs
# Update the plugin

### What does this PR do?

Except for the HTML template functionality, I basically rewrote the entire thing:

**Babel to the rescue!**

Instead of using the [`ng-annotate-loader`](https://yarnpkg.com/en/package/ng-annotate-loader) for webpack, it now uses the [`angularjs-annotate`](https://yarnpkg.com/en/package/babel-plugin-angularjs-annotate) plugin for Babel:

1. The loader is outdated.
2. A generic solution like this is easier to maintain.
3. projext already runs Babel, so now the transformation happens during the Babel process and not after it.
4. Thanks to the new [Babel helper](https://github.com/homer0/projext/pull/38), the logic that adds support for the annotation has been reduced to two small calls.

**EVERYTHING is _"private/protected"_ now**

I'm doing this on several plugins and projext services now: Methods and properties that weren't intended to be used outside its class are now prefixed with an underscore (`_`), to mark them as _"protected/private"_.

Sometimes I consider some methods/properties to be "ok" as _"public"_ as they may be of utility for someone, but it wasn't until a few releases ago that I realized that by thinking that, changing those methods/properties, would require a breaking release, when there was no "real" reason for those things to be _"public"_.

So, from now on, use events and reducers and try to stay away from methods and properties prefixed with underscore.

**Dependencies up to date**

All production and development dependencies were updated to their latest version.

Now, there's a special mention: ESLint. It latest version is `v5`, but the Airbnb preset I use is still not compatible with it, so I updated ESLint to the last version before `v5`: `v4.19.1`.

I also updated a few things on the `.eslintrc` after getting the Airbnb preset last version:

- `lines-between-class-members`: I disabled this rule because I already fill the space between methods with comments, so I see not reason for it.
- `operator-linebreak`: I changed the default value from the Airbnb, `before`,  to `after`. The convention I use and which I believe is easier to read is with the operators at the end of the line.

**`hooks` instead of `install-hooks`**

The `yarn`/`npm` script to install the repositories hook has been changed to just `hooks`.

### How should it be tested manually?

This shouldn't be breaking, so try to build your AngularJS app using this branch, it should be fine.

And of course...

```bash
npm test
# or
yarn test
```
